janusgraph-lucene/src/main/java/org/janusgraph/diskstorage/lucene/LuceneIndex.java

```diff
@@ -498,58 +515,134 @@ private static Query numericQuery(String key, Cmp relation, Number value) {
         }
     }
 
-    private SearchParams convertQuery(Condition<?> condition, KeyInformation.StoreRetriever information) {
-        SearchParams params = new SearchParams();
+    // adapted from SolrIndex
+    private List<String> customTokenize(Analyzer analyzer, String fieldName, String value) {
+        final List<String> terms = new ArrayList<>();
+        try (CachingTokenFilter stream = new CachingTokenFilter(analyzer.tokenStream(fieldName, value))) {
+            final TermToBytesRefAttribute termAtt = stream.getAttribute(TermToBytesRefAttribute.class);
+            stream.reset();
+            while (stream.incrementToken()) {
+                terms.add(termAtt.getBytesRef().utf8ToString());
+            }
+            return terms;
+        } catch (IOException e) {
+            throw new IllegalArgumentException(e.getMessage(), e);
+        }
+    }
+
+    private void tokenize(SearchParams params, final Mapping mapping, final LuceneCustomAnalyzer delegatingAnalyzer, String value, String key, JanusGraphPredicate janusgraphPredicate) {
+        final List<String> terms;
+        if (mapping == null) {
+            terms = Text.tokenize(value);
+        } else {
+            final Analyzer analyzer = delegatingAnalyzer.getWrappedAnalyzer(key);
+            if (analyzer != null) {
```

Comment:
It looks like based on the current implementation `getWrappedAnalyzer` will always return a non-null analyzer. Is that intended?

Reply:
No, there is something wrong here. I think that in the last refactoring pass, I messed up the choice of analyzers while symplifying things (String and TEXT fields should still be able to use the janusgraph (non-lucene) way of computing terms) . Looking into it...

Reply:
simplified this further. Thanks :)